aesd-char-driver/main.c

```diff
@@ -308,13 +331,53 @@ static loff_t aesd_llseek(struct file *file, loff_t offset, int whence)
 
     if( new_buf_position < 0 ) new_buf_position = 0;
 
+    PDEBUG("aesdchar: aesd_llseek: f_pos = %lld, offset = %lld new_buf_position =%lld \n", file->f_pos, offset, new_buf_position);
     file->f_pos = new_buf_position;
 
     mutex_unlock(&dev_struct->aesdchar_mutex);
 
     return new_buf_position;
 }
 
+static long aesd_ioctl(struct file *filep, unsigned int cmd, unsigned long arg)
+{
+    struct aesd_seekto st;
+    long err = 0;
+    size_t entry_offset_byte_rtn = 0;
+    struct aesd_dev *dev_struct = (struct aesd_dev *)filep->private_data;
+    PDEBUG("aesd_ioctl \n");
+    memset(&st, 0x0, sizeof(struct aesd_seekto));
+    err = mutex_lock_interruptible(&dev_struct->aesdchar_mutex);
```

Comment:
@sujoyray-ucolorado is this the minimum locking interval?  Does `copy_from_user` need to be within the locking interval?